src/tests/ftest/util/ServerUtils.py

```
@@ -33,18 +34,110 @@
+    try:
+        with open('{}/{}'.format(basepath,
+                                 DEFAULT_YAML_FILE), 'r') as read_file:
+            default_value = yaml.safe_load(read_file)
+    except Exception as excpn:
+        print("<SERVER> Exception occurred: {0}".format(str(excpn)))
+        traceback.print_exception(excpn.__class__, excpn, sys.exc_info()[2])
+        raise ServerFailed("Failed to Read {}/{}".format(basepath,
+                                                         DEFAULT_YAML_FILE))
+
+    #Read the values from avocado_testcase.yaml file if test ran with Avocado.
+    avocado_yaml_value = ""
+    if "AVOCADO_TEST_DATADIR" in os.environ:
+        avocado_yaml_file = str(os.environ["AVOCADO_TEST_DATADIR"]).\
+                                split(".")[0] + ".yaml"
+        avocado_yaml_file_tmp = '{}.tmp'.format(avocado_yaml_file)
+
+        # Yaml Python module is not able to read !mux so need to
+        # remove !mux before reading.
+        remove_mux_from_yaml(avocado_yaml_file, avocado_yaml_file_tmp)
+
+        # Read avocado test yaml file.
+        try:
+            with open('{}'.format(avocado_yaml_file_tmp), 'r') as read_file:
+                avocado_yaml_value = yaml.safe_load(read_file)
+        except Exception as excpn:
+            print("<SERVER> Exception occurred: {0}".format(str(excpn)))
+            traceback.print_exception(excpn.__class__, excpn, sys.exc_info()[2])
+            raise ServerFailed("Failed to Read {}"
+                               .format('{}.tmp'.format(avocado_yaml_file)))
+        #Remove temporary file
+        os.remove(avocado_yaml_file_tmp)
+    #Update values from avocado_testcase.yaml in DAOS yaml variables.
+    for key in avocado_yaml_value['server_config']:
+        if key in default_value['servers'][0]:
+            default_value['servers'][0][key] = avocado_yaml_value\
+            ['server_config'][key]
+        elif key in default_value:
```

What if neither of these branches is true?  I.e. the key isn't in the global values or the `server` values in `default_value`.

Should we raise an error here about a bad file format, or is it OK for neither of these to exist?

---

It will update the new value if it's exist and if it's not it's OK to pass through.